Tests/BinaryCompatibility.lean

```diff
@@ -0,0 +1,207 @@
+    )",
+    "(module
+      (func (param $x i32) (param i32) (result i32) (i32.add (i32.const 40) (i32.const 2)))
+    )",
+    "(module
+        ( func (param $x i32) (param i32) )
+      )",
+    "(module
+        (func (param $x i32))
+      )",
+    "(module
+        ( func )
+      )",
+    "(module
+      (func $main (export \"main\")
+        (param $x i32)
+        (param i32)
+        (result i32 i32)
+
+        (block (result i32) (i32.const 3) (i32.add (br 0) (i32.const 9)))
+        (i32.add
+          (i32.const 1499550000)
+          (i32.add (i32.const 9000) (i32.const 17))
+        )
+
+      )
+    )",
+    "(module
+      (func $main (export \"main\")
+        (param $x i32)
+        (param i32)
+        (result i32 i32)
+
+        (block (result i32) (i32.const 3) (i32.add (br 0) (i32.const 9)))
+        (i32.add
+          (i32.const 1499550000)
+          (i32.add (i32.const 17) (i32.const 9000))
+        )
+
+      )
+    )"
+]
+
+-- TODO: Prove termination because lengths decrease 1 at a time.
+-- Byte by byte, compare the two variables, reporting errors.
+partial def loop (rs : ByteArray) (os : ByteArray) : Bool :=
+  dbg_trace s!"loop:\n{rs}\n{os}\n***********************************"
+  let il := os.data.toList
+  let jl := rs.data.toList
+  -- Let's see if the lengths aren't the same. If they aren't, we're done.
+  match il.length != jl.length with
+  | true => false
+  | false => Id.run $ do
+      -- If they are, let's compare the two lists.
+      match (il, jl) with
+      | ([], []) => true
+      | (i::is, j::js) => do
+        if i ≠ j then
+          false
+        else
+          loop (ByteArray.mk $ Array.mk js) (ByteArray.mk $ Array.mk is)
+      | _ => false
+
+/- Generic Wasm module test that uses loop to check for stuff. -/
+partial def testGeneric : String → ByteArray → ByteArray → TestSeq
+  | mod, rs, os => test s!"Binary representation is compatible with {fname mod}\n=== CODE ===\n{mod}\n||| END OF CODE |||" $ loop rs os
+
+/- Here's how Main used to test a particular module:
+
+  IO.println s!"{i} is represented as:"
+  let o_parsed_module ← parseTestP moduleP i
+  match o_parsed_module.2 with
+  | .error _ => IO.println "FAIL"
+  | .ok parsed_module => do
+    IO.println s!">>> !!! >>> It is converted to bytes as: {mtob parsed_module}"
+    let f := System.mkFilePath ["/tmp", "mtob.41.wasm"]
+    let h ← IO.FS.Handle.mk f IO.FS.Mode.write
+    h.write $ mtob parsed_module
+    gO.println "It's recorded to disk at /tmp/mtob.41.wasm"
+
+Instead of this, here, we're going to:
+
+ 1. Make the reference WASM bytes with `wasm-sandbox` invocation, as provided by function `w2b`.
+ 2. Read the reference bytes into a variable.
+ 3. Compile our bytes and read those into a variable.
+ 4. Write our bytes into a file called s!"{fname}.lean".
+ 5. Byte by byte, compare the two variables, reporting errors.
+-/
+partial def moduleTestIO (x : String) : IO UInt32 := do
```

Comment:
I kinda don't like how low-level this is, considering LSpec is really awesome about monadic testing. Why not use [LSpec capabilities](https://github.com/yatima-inc/LSpec/blob/main/LSpec/LSpec.lean#L265) and create a `IO TestSeq` instead of barebones `IO UInt32`?

Reply:
I don't know! I'm just glad I can [run tests at all](https://github.com/yatima-inc/Wasm.lean/commit/21459e2558f426bee3e4c9749d944e4ed05f413f)

Reply:
But yeah, to explain how I ended up here.

I need to write tests in one place, but I also want to have effectful stuff to happen, like running a system binary. To do this I need some sort of test generator and my own test runner, as far as I understand.